src/Header/MainManager.h

```diff
@@ -0,0 +1,43 @@
+// Copyright (c) 2021, 鍾淯丞, 周杰仕, 林仁鴻. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#ifndef Group4_Header_MainManager_H_
+#define Group4_Header_MainManager_H_
+
+#include <netinet/in.h>
+
+#include <vector>
+#include <unordered_map>
+
+#include "HttpListener.h"
+#include "RequestHandler.h"
+
+
+
+//will initialize http listener, do multiprocess when listener callback
+class MainManager
+{
+public:
+    //httpListener callback, do fork and new requestHandler to call api, child process add to parents vector for managing.
+    void NewRequest(const string &requestStr);
+    //requestHandler callback, remove process from parents vector.
+    void EndRequest();
+    //for managing all processes.
+    std::unordered_map<struct sockaddr_in, pid_t> socketMap;
+    //singleton code convention method, listen port number is required.
+    static MainManager* GetInstance(const int &portNum);
```

Comment:
Please add a new InitInstance() for singleton that requires initialization.
You can assert() if GetInstance() is called without InitInstance.

Reply:
Also, use uint16_t for port number.

Reply:
Done